en/docs/develop/customizations/creating-new-connector.md

```diff
@@ -41,54 +41,192 @@ The `org.wso2.integration.connector.googlebooks` directory is now created with a
+
+`config/init.xml`
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<template name="init" onError="fault" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter name="baseUrl" description="Base URL of the endpoint"/>
+    <sequence>
+        <property name="uri.var.baseUrl" expression="$func:baseUrl" />
+    </sequence>
+</template>
+```
 
-3.  Create a file named `component.xml` in the `googlebooks_volume` directory and add the following content:
-    ```xml
-    <?xml version="1.0" encoding="UTF-8"?>
-    <component name="googlebooks_volume" type="synapse/template">
-        <subComponents>
-            <component name="listVolume">
+2. The `functions` directory contains the operation definitions. The template provides a sample operation; update it to the `listVolume` operation. 
+
+`functions/component.xml`
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<component name="functions" type="synapse/template" >
+    <subComponents> 
+        <component name="listVolume" >
                 <file>listVolume.xml</file>
+                <displayName>List volumes</displayName>
                 <description>Lists volumes that satisfy the given query.</description>
-            </component>
-        </subComponents>
-    </component>
-    ```
+        </component>			
+    </subComponents>    
+</component>
+```
 
-4.  Edit the `connector.xml` file in the `src/main/resources` directory and replace the contents with the following dependency:
-    ```xml
-    <?xml version="1.0" encoding="UTF-8"?>
-    <connector>
-        <component name="googleBooks" package="org.wso2.integration.connector" >
-          <dependency component="googlebooks_volume"/>
-            <description>wso2 sample connector library</description>
-        </component>
-    </connector>
-    ```
+Rename `functions/sample_operation.xml` to `functions/listVolume.xml` and update its content as follows.
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<template xmlns="http://ws.apache.org/ns/synapse" name="listVolume">
+    <parameter name="searchQuery" description="Full-text search query string." />
+	<parameter name="responseVariable" description="The name of the variable to which the output should be stored."/>
+	<parameter name="overwriteBody" description="Replace the Message Body in Message Context with the response of the operation."/>
+    <sequence>
+        <property name="uri.var.searchQuery" expression="$func:searchQuery" />
+        <call>
+            <endpoint>
+                <http method="get" uri-template="{uri.var.baseUrl}/books/v1/volumes?q={uri.var.searchQuery}" />
+            </endpoint>
+        </call>
+    </sequence>
+</template>
+```
 
-5. Create a folder named `icon` in the `/src/main/resources` directory and add two icons.
+3. Update the output schema located in the `outputSchema` directory. Rename `outputSchema/sample_operation.json` to `outputSchema/listVolume.json` and update its content as follows.
+
+```json
```

Comment:
If we have an idea about the payload structure of the google list operaton, then maybe we can update that here. wdyt?